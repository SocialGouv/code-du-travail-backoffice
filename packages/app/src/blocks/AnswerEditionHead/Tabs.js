import styled from "@emotion/styled";
import React from "react";
import { Flex } from "rebass";

import AnswerEditionHeadBlockTab from "./Tab";

const Container = styled(Flex)`
  margin-top: 0.5rem;
`;

export const TABS = {
  EDITOR: 0,
  REFERENCES: 1,
};

const AnswerEditionHeadBlockTabs = ({ currentTab, onChange, referencesCount }) => (
  <Container>
    <AnswerEditionHeadBlockTab
      isActive={currentTab === TABS.EDITOR}
      onClick={() => onChange(TABS.EDITOR)}
    >
      Édition
    </AnswerEditionHeadBlockTab>
    <AnswerEditionHeadBlockTab
      isActive={currentTab === TABS.REFERENCES}
      onClick={() => onChange(TABS.REFERENCES)}
    >
      {`Références juridiques (${referencesCount})`}
    </AnswerEditionHeadBlockTab>
  </Container>
);

export default AnswerEditionHeadBlockTabs;
