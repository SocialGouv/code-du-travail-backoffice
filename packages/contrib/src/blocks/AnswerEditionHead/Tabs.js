import React from "react";
import { Flex } from "rebass";
import styled from "styled-components";

import Tab from "./Tab";

const Container = styled(Flex)`
  margin-top: 0.5rem;
`;

export const TABS = {
  EDITOR: 0,
  REFERENCES: 1
};

export default ({ currentTab, onChange }) => (
  <Container>
    <Tab
      icon="pen"
      isActive={currentTab === TABS.EDITOR}
      onClick={() => onChange(TABS.EDITOR)}
    >
      Édition du contenu
    </Tab>
    <Tab
      icon="book"
      isActive={currentTab === TABS.REFERENCES}
      onClick={() => onChange(TABS.REFERENCES)}
    >
      Références juridiques
    </Tab>
  </Container>
);
