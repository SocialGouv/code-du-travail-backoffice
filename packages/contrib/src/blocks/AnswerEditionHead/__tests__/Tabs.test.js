import React from "react";
import { fireEvent, render } from "@testing-library/react";

import Tabs, { TABS } from "../Tabs";

describe.skip("[Contrib] blocks/AnswerEditionHead/<Tabs />", () => {
  const props = {
    currentTab: TABS.EDITOR,
    onChange: jest.fn(),
    referencesCount: 2,
    tagsCount: 12
  };

  const { container, getByText } = render(<Tabs {...props} />);

  it("should match snapshot", () => {
    expect(container).toMatchSnapshot();
  });

  it("should trigger onChange() with the expected param (0)", () => {
    fireEvent.click(getByText("Édition"));

    expect(props.onChange).toHaveBeenCalledWith(TABS.EDITOR);
  });

  it("should trigger onChange() with the expected param (1)", () => {
    fireEvent.click(getByText(/Références juridiques \(\d+\)/));

    expect(props.onChange).toHaveBeenCalledWith(TABS.REFERENCES);
  });

  it("should trigger onChange() with the expected param (2)", () => {
    fireEvent.click(getByText(/Étiquettes \(\d+\)/));

    expect(props.onChange).toHaveBeenCalledWith(TABS.TAGS);
  });
});
