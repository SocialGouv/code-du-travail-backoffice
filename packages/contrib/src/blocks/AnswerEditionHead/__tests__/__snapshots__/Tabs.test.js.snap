// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Contrib] blocks/AnswerEditionHead/<Tabs /> should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa sc-bwzfXH Tabs__Container-sc-10g095d-0 eUHfiE"
  >
    <div
      class="sc-bdVaJa sc-bwzfXH Tab__Container-sc-9u46t0-0 jSKpAZ"
    >
      Édition
    </div>
    <div
      class="sc-bdVaJa sc-bwzfXH Tab__Container-sc-9u46t0-0 hyrWDR"
    >
      Références juridiques (2)
    </div>
    <div
      class="sc-bdVaJa sc-bwzfXH Tab__Container-sc-9u46t0-0 hyrWDR"
    >
      Étiquettes (12)
    </div>
  </div>
</div>
`;
